pkg/graveler/staging/manager.go

```diff
@@ -10,15 +10,24 @@ import (
 )
 
 type Manager struct {
-	store kv.StoreMessage
-	log   logging.Logger
+	store  kv.StoreMessage
+	log    logging.Logger
+	wakeup chan bool
+
+	// notifiers are being called with every successful cleanup cycle
+	notifiers []chan bool
 }
 
-func NewManager(store kv.StoreMessage) *Manager {
-	return &Manager{
-		store: store,
-		log:   logging.Default().WithField("service_name", "staging_manager"),
+func NewManager(ctx context.Context, store kv.StoreMessage, notifiers ...chan bool) *Manager {
```

Comment:
Notifiers are for testing and to make the implementation of this manager simple I'll just add a method that can set callback func to notify on drop completed.
This way we can set a function that will close a channel in the case of our test and we can use it for logging if we like in non-test.